blaze-client/src/main/scala/org/http4s/client/blaze/PoolManager.scala

```diff
@@ -1,66 +1,146 @@
 package org.http4s.client.blaze
 
+import java.util
+import java.util.concurrent.atomic.{AtomicBoolean, AtomicInteger}
+
 import org.http4s.Request
-import org.http4s.Uri.{Authority, Scheme}
 import org.log4s.getLogger
 
 import scala.collection.mutable
-import scalaz.concurrent.Task
-
-
-/* implementation bits for the pooled client manager */
-private final class PoolManager (maxPooledConnections: Int, builder: ConnectionBuilder) extends ConnectionManager {
+import scalaz.{-\/, \/-, \/}
+import scalaz.syntax.either._
+import scalaz.concurrent.{Actor, Task}
 
-  require(maxPooledConnections > 0, "Must have finite connection pool size")
+private object PoolManager {
+  type Callback[A] = Throwable \/ A => Unit
+  sealed trait Protocol
```

Comment:
I think it would be better to define these in the class: they shouldn't be shared between different pool instances and since they are all classes we don't save allocations by caching case objects.